### ISSUE TITLE
Multiple checks in a row. Promise all and array.map

### ISSUE THREAD

**Issue by vaidas-lungis** [AliasIO/Wappalyzer#1692]
Can somebody assist me with the JS code
I want to provide a list of domains to wappalyzer and collect responses. But It only returns last items `data` and looks like not all promises resolve. 

test.js
```js
var wappalyzer = require('./node_modules/wappalyzer/driver');

var domains = ['https://hostinger.co.uk', 'https://hostinger.com'];

wappalyzer.quiet = false;

Promise.all(domains.map(function (item){
    return wappalyzer.analyze(item)
        .then(json => {
            console.log('FINISHED PROMISE: '+item);
        })
        .catch(error => {
            console.error('ERROR'+error);
        });
})).then(function(values){
    console.log('FINISHED ALL'+values)
});

```

console output
```console
root@10af9694b853:/directory# node -v
v8.2.1
root@10af9694b853:/directory# npm -v 
5.3.0
root@10af9694b853:/directory# node test.js 
[wappalyzer debug] [core] 5 apps detected: AngularJS, Facebook, Google Analytics, Modernizr, jQuery on https://hostinger.co.uk
FINISHED PROMISE: https://hostinger.com
[wappalyzer debug] [core] 5 apps detected: AngularJS, Facebook, Google Analytics, Modernizr, jQuery on https://hostinger.com
^C
```

Apps collected from both sources but only one output of `displayApps` :/

What am I missing here?

**Comment by AliasIO** [comment]
You need to return a promise in the map function, this should work:

```javascript
Promise.all(domains.map(domain => wappalyzer.analyze(domain))
  .then(results => console.log(results));
```

**Comment by vaidas-lungis** [comment]
Yes, but by providing an array with two or more items `Promise.all` does not fire as it waits for all promises to be resolved. Somehow [resolve(apps); @ driver.js:54]( https://github.com/AliasIO/Wappalyzer/blob/f0654e5bc0f90374362ff4f094ba73190dccd631/src/drivers/npm/driver.js#L54) fails to resolve promise.

**Comment by AliasIO** [comment]
You're right, the issue is that the driver only uses a single instance of `wappalyzer`. I've changed it to a class, which works.

Fixed in v5.0.7: 3057e396d9ddcddb80b23d57689cd706cfd90659.
